### ISSUE TITLE
Cannot set socket options in helm chart since 1.7.0

### ISSUE THREAD

**Issue by ctomc** [solo-io/gloo#4551]
**Describe the bug**
We are trying to configure socket options as described in https://docs.solo.io/gloo-edge/master/guides/dev/configuring-socket-options/
It worked fine for us in 1.6.x versions of gloo, but in 1.7 helm template fails.

**To Reproduce**
set values.yaml for chart as: 
```yaml
gatewayProxies:
  gatewayProxy:
    gatewaySettings:
      options:
        socketOptions:
          - description: "enable keep-alive"
            level: 1
            name: 9
            int_value: 1
            state: STATE_LISTENING
          - description: "idle time before first keep-alive probe is sent"
            level: 6
            name: 4
            int_value: 60
            state: STATE_LISTENING
          - description: "keep-alive interval" # TCP protocol
            level: 6
            name: 5
            int_value: 60
            state: STATE_LISTENING
```
> helm upgrade gloo glooe/gloo-ee --namespace gloo-system -f values.yaml --version=1.7.0

results in 

> Error: UPGRADE FAILED: YAML parse error on gloo-ee/charts/gloo/templates/8-default-gateways.yaml: error converting YAML to JSON: yaml: line 13: did not find expected key

**Expected behavior**
helm upgrade succeeds.

**Additional context**
Add any other context about the problem here, e.g.
- Gloo Edge version: old one 1.6.15-1.6.22, broken that we tried was 1.7.0
- Kubernetes version, 1.17, 1.18, 1.19
- helm 3.5.2

**Comment by sam-heilbron** [comment]
Hi @ctomc, are you still facing this issue? I was able to execute `helm upgrade -i gloo glooe/gloo-ee --version=1.7.0 --namespace gloo-system -f values.yaml` without any issues.

**Comment by sam-heilbron** [comment]
Hi @ctomc a fix for this was released in gloo OSS v1.7.1 (https://github.com/solo-io/gloo/releases/tag/v1.7.1). This should get picked up in the next Gloo Enterprise release

**Comment by ctomc** [comment]
Hey, just tried with enterprise 1.7.1 and can still reproduce it. But but it looks like it fails on different part of template

> Error: UPGRADE FAILED: YAML parse error on gloo-ee/charts/gloo/templates/8-default-gateways.yaml: error converting YAML to JSON: yaml: line 40: did not find expected key

**Comment by ctomc** [comment]
@sam-heilbron found the workaround or combination of the issue. It only fails if also access logging is configured.

so you have something like this:

 ```
gatewayProxies:
    gatewayProxy:
      gatewaySettings:
        options:
          socketOptions:
            - description: "keep-alive interval" # TCP protocol
              level: 6
              name: 5
              int_value: 60
              state: STATE_LISTENING
        accessLoggingService:
          accessLog:
            - fileSink:
                path: /dev/stdout
                stringFormat: ""
```

of only socetOptions are present if works fine.

**Comment by sam-heilbron** [comment]
@ctomc If socket options are not configured, and just access logging is configured does the issue exist? Is it a problem with access logging or the existence of both at the same time?

**Comment by ctomc** [comment]
@sam-heilbron it only happens when both access logging & gatewaysettings are set, if only one of them is configured it works fine.

**Comment by sam-heilbron** [comment]
Hi @ctomc I believe this has been resolved by the above PR and is available in Gloo Edge 1.8.5.
